src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -331,6 +331,33 @@ partition_downloader::download_log(const remote_manifest_path& manifest_key) {
     co_return result;
 }
 
+void partition_downloader::update_downloaded_offsets(
+  std::vector<partition_downloader::offset_range> dloffsets,
+  partition_downloader::download_part& dlpart) {
+    std::sort(dloffsets.begin(), dloffsets.end());
+    for (auto it = dloffsets.rbegin(); it != dloffsets.rend(); it++) {
+        auto offsets = *it;
+        if (dlpart.range.min_offset < dlpart.range.max_offset) {
+            // This will be triggered for every iteration except the first one
+            auto expected = offsets.max_offset + model::offset(1);
+            if (dlpart.range.min_offset != expected) {
+                vlog(
+                  _ctxlog.warn,
+                  "Gap detected at {}, restored offset range {}-{}",
+                  expected,
+                  dlpart.range.min_offset,
+                  dlpart.range.max_offset);
+                break;
```

Comment:
I may be misunderstanding, but how will this adjust the offsets? Looks like this code simply ignores the offset ranges after the gap.

Reply:
it adjusts starting offset by ignoring everything after the gap